### PR TITLE
feat(api-client,core): Send and receive typing notifications

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: packages/core/src/main/Account.ts [hunk 1]
```diff
@@ -238,46 +239,45 @@ class Account extends EventEmitter {
...
-          });
-          break;
-        }
-      }
-    });
+    const sessionId = CryptographyService.constructSessionId(otrMessage.from, otrMessage.data.sender);
+    const ciphertext = otrMessage.data.text;
+    const decryptedMessage = await this.service.cryptography.decrypt(sessionId, ciphertext);
+    const genericMessage = this.protocolBuffers.GenericMessage.decode(decryptedMessage);
+    console.log('genericMessage', genericMessage);
```
Comment (lipis) [inline packages/core/src/main/Account.ts]: do we allow `console.log` around here?

Comment (ffflorian) [reply packages/core/src/main/Account.ts]: No, will remove it.

File: packages/core/src/demo/receiver.js [hunk 2]
```diff
@@ -59,6 +59,20 @@ const {MemoryEngine} = require('@wireapp/store-engine/dist/commonjs/engine');
     await account.service.conversation.sendPing(conversationId);
   });
 
+  account.on(Account.INCOMING.TYPING, async data => {
+    const {
+      conversation: conversationId,
+      from,
+      data: {status},
+    } = data;
+    console.log(`Typing in "${conversationId}" from "${from}".`, data);
```
Comment (lipis) [inline packages/core/src/demo/receiver.js]: there are more :)

File: packages/core/src/demo/receiver.js [hunk 3]
```diff
@@ -59,6 +59,20 @@ const {MemoryEngine} = require('@wireapp/store-engine/dist/commonjs/engine');
...
+    if (status === 'started') {
+      await account.service.conversation.sendTypingStart(conversationId);
+    } else {
+      await account.service.conversation.sendTypingStop(conversationId);
+    }
+  });
+
   try {
     console.log('Logging in ...');
```
Comment (lipis) [inline packages/core/src/demo/receiver.js]: not part of this PR.. but maybe it should go too..

Comment (ffflorian) [reply packages/core/src/demo/receiver.js]: No, this is fine for the demo.

File: packages/core/src/demo/receiver.js (hunk 2 above)
Comment (ffflorian) [reply packages/core/src/demo/receiver.js]: Fine for demo
